roles/rsyslog/vars/main.yml

```diff
@@ -1,3 +1,8 @@
+# __rsyslog_enabled
+#
+# Enable or disable ``rsyslog`` management.
+__rsyslog_enabled: true
```

Comment:
@nhosoi Hi, why did you move this to vars? Is it always enabled? And same question for the 
__rsyslog_system_log_dir: '/var/log' below.

I believe they should be defaults so that they can be updated.no?

Reply:
> @nhosoi Hi, why did you move this to vars? Is it always enabled? And same question for the
> __rsyslog_system_log_dir: '/var/log' below.
> 
> I believe they should be defaults so that they can be updated.no?

The motivation is turning them (`rsyslog_enabled, rsyslog_system_log_dir`) to the internal variables (by starting them with '__' and moving them to the vars files) and only allowing to set them via corresponding the external variables.  For instance, use `logging_enabled` for `__rsyslog_enabled`; `logging_system_log_dir` for `__rsyslog_system_log_dir`.

Having them set in the vars file, I believe the values are just the default values.  So, __rsyslog_enabled is true, by default, and you can set logging_enabled to false to disable it.  But it was a very good point, @sradco.  All of the CI tests are setting logging_enabled to true.  They are not needed.  And I'm not testing `logging_enabled: false`.  I'm adding it.  Thanks!

Reply:
This is the `logging` role - whether to use `rsyslog` or not is supposed to be an implementation detail, so most of the parameters exposed to end users of the `logging` role will be `logging_*` parameters.  Of course, there will be a need for users to supply provider specific parameters e.g. as a user of the logging role, I know for a fact that I want to use `rsyslog`, and I must supply some `rsyslog` specific parameters.

Reply:
> This is the `logging` role - whether to use `rsyslog` or not is supposed to be an implementation detail, so most of the parameters exposed to end users of the `logging` role will be `logging_*` parameters. Of course, there will be a need for users to supply provider specific parameters e.g. as a user of the logging role, I know for a fact that I want to use `rsyslog`, and I must supply some `rsyslog` specific parameters.

Thanks, @richm.  That is correct.  This is going to be the minimum configuration - input from imjournal, output to the local files.  No rsyslog indication in it. (of course, this is because logging_collector is set to rsyslog in the defaults/main.yml, though. ;)
```
      vars:
        logging_collections:
          - name: basic_input
            type: basics
```

Reply:
@nhosoi 

> Having them set in the vars file, I believe the values are just the default values. So, __rsyslog_enabled is true, by default, and you can set logging_enabled to false to disable it.

No, variables set in vars are not default values. They are very high in the priority list, and it is less easy to override them. This is mentioned here: https://github.com/oasis-roles/meta_standards/blob/master/README.md#vars-vs-defaults, specifically: "Avoid giving default values in vars/main.yml as such values are very high in the precedence order and are difficult for users and consumers of a role to override." For details on variable precedence, see https://docs.ansible.com/ansible/latest/user_guide/playbooks_variables.html#variable-precedence-where-should-i-put-a-variable . Note that defaults are on level 2, while role vars are on level 15. It may be the case that the way you are overriding them actually works, still, to conform to the best practices and reduce confusion, they should be given in defaults IMO.
Can you perhaps better explain the motivation for moving them from defaults to vars, please?